python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -582,25 +561,41 @@ def _visit_level(base_dir, level, part_keys):
                 str(base_dir),
                 '{}={}'.format(name, value)
             ])
-            fs.mkdir(level_dir)
+            try:
+                fs.create_dir(level_dir)
+            except AttributeError:
+                fs.mkdir(level_dir)
```

Comment:
Is this `except` case still needed?

Reply:
I attempted a different approach to fix the failing tests. As we discussed some time ago: instead of `try`-`except` I used PyArrow's `S3FileSystem` in the `s3_example_s3fs`. I also corrected a bug in `s3_example_fs` where `allow_bucket_creation=True `was missing.

Commit: https://github.com/apache/arrow/pull/39825/commits/142af852c9caa821061bdf650a241c377319d57d